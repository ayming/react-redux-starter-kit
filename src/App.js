import React, { Component } from 'react'

import logo from './logo.svg'
import s from './App.scss'

class App extends Component {
  render() {
    return (
      <div className={s.wrapper}>
        <header className={s.header}>
          <img src={logo} className={s.logo} alt="logo" />
          <h1 className={s.title}>Welcome to React</h1>
        </header>
        <p className={s.intro}>
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    )
  }
}

export default App
