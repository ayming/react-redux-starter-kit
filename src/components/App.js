import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

const contextType = {
  pathname: PropTypes.string.isRequired,
  query: PropTypes.shape({}),
}

class App extends PureComponent {
  static propTypes = {
    context: PropTypes.shape(contextType).isRequired,
    children: PropTypes.element.isRequired,
  }

  static childContextTypes = contextType

  getChildContext() {
    const { context } = this.props
    return context
  }

  render() {
    const { children } = this.props
    // NOTE: If you need to add or modify header, footer etc. of the app,
    // please do that inside the Layout component.
    return React.Children.only(children)
  }
}

export default App
