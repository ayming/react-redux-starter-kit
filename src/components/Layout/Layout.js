import React, { PureComponent, Fragment } from 'react'
import PropTypes from 'prop-types'

import Header from '../Header'
import './Layout.scss'

class Layout extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired,
  }

  render() {
    const { children } = this.props
    return (
      <Fragment>
        <Header />
        {children}
      </Fragment>
    )
  }
}

export default Layout
