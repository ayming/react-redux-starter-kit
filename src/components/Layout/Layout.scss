@import '../variables.scss';

/*
 * Base styles
 * ========================================================================== */

html {
  color: #222;
  font-weight: 100;
  font-size: 1em; /* ~16px; */
  font-family: $font-family-base;
  line-height: 1.375; /* ~22px */
}

body {
  margin: 0;
}
