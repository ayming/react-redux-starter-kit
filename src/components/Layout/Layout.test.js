/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../App'
import Layout from './Layout'
import noop from '../../utils/noop'

describe('Layout', () => {
  test('renders children correctly', () => {
    const wrapper = renderer
      .create(
        <App
          context={{
            pathname: '',
            injectReducer: noop,
            store: {
              dispatch: noop,
              getState: noop,
              subscribe: noop,
            },
          }}
        >
          <Layout>
            <div className="child" />
          </Layout>
        </App>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
