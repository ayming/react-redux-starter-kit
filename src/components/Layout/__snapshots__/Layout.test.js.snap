// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders children correctly 1`] = `
Array [
  <header
    className="wrapper"
  >
    <a
      href="/"
      method="push"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo"
        src="logo.svg"
      />
    </a>
    <h1
      className="title"
    >
      Welcome to React
    </h1>
  </header>,
  <div
    className="child"
  />,
]
`;
